File: Selenium/Activities/Activity17.java
package examples;

import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import io.github.bonigarcia.wdm.WebDriverManager;

public class Activity17 {

    public static void main(String[] args) {

        WebDriverManager.firefoxdriver().setup();

        WebDriver driver = new FirefoxDriver();

        driver.get("https://training-support.net/webelements/selects");

        System.out.println("Page title: " + driver.getTitle());

        WebElement dropdown = driver.findElement(By.cssSelector("select.h-80"));

        Select multiSelect = new Select(dropdown);

        multiSelect.selectByVisibleText("HTML");

        multiSelect.selectByIndex(3);
        multiSelect.selectByIndex(4);
        multiSelect.selectByIndex(5);

        multiSelect.selectByValue("nodejs");

        System.out.println("\nSelected options:");
        List<WebElement> selected = multiSelect.getAllSelectedOptions();
        for (WebElement option : selected) {
            System.out.println(option.getText());
        }

        multiSelect.deselectByIndex(4);

        System.out.println("\nAfter deselecting index 4:");
        selected = multiSelect.getAllSelectedOptions();
        for (WebElement option : selected) {
            System.out.println(option.getText());
        }

        driver.quit();
    }
}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select

driver = webdriver.Firefox()

driver.get("https://training-support.net/webelements/selects")

print("Page title:", driver.title)

dropdown = driver.find_element(By.CSS_SELECTOR, "select.h-80")
multi_select = Select(dropdown)

multi_select.select_by_visible_text("HTML")

multi_select.select_by_index(3)
multi_select.select_by_index(4)
multi_select.select_by_index(5)

multi_select.select_by_value("nodejs")

print("\nSelected options:")
for option in multi_select.all_selected_options:
    print(option.text)

multi_select.deselect_by_index(4)

print("\nAfter deselecting index 4:")
for option in multi_select.all_selected_options:
    print(option.text)

driver.quit()